### PR TITLE
8 feature 8 chat

### PR DESCRIPTION
**Pull request by Jennifer-xxx**
Implemented the chat feature based on clean architecture using MVP. The branch overall functions well. Users are able to chat with each other through the pop-up window. All chat data is saved into and loaded from MessageManagers.ser if there is any. Tests are to be added by Alyssa.

### CODE REVIEW

File: src/main/java/entity/UserList.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package entity;
+
```
Comment (Christina0313) [inline src/main/java/entity/UserList.java:2]: This class is no longer used.

File: src/main/java/entity/UserListIterator.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package entity;
```
Comment (Christina0313) [inline src/main/java/entity/UserListIterator.java:1]: This class is no longer used.

File: src/main/java/entity/UserPreference.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package entity;
```
Comment (Christina0313) [inline src/main/java/entity/UserPreference.java:1]: This file is no longer used.

File: src/main/java/entity/Users.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package entity;
```
Comment (Christina0313) [inline src/main/java/entity/Users.java:1]: This class is no longer used.

File: src/main/java/gui/MainFrame.java [hunk 5]
```diff
@@ -15,9 +14,9 @@ public MainFrame() {
         boolean isLoggedIn = controller.isUserLoggedin();
         System.out.println(isLoggedIn);
         // if user is logged in
-        if(isLoggedIn) {
+        if (isLoggedIn) {
             this.getContentPane().removeAll();
-            this.add(new MainPanel(this));
+            this.add(new MainPanel());
```
Comment (Christina0313) [inline src/main/java/gui/MainFrame.java:19]: The main frame is different from main. Please double check

File: src/main/java/gui/MainPanel.java [hunk 6]
```diff
@@ -1,10 +1,12 @@
 package gui;
+import gui.chat.FrontPagePanel;
+
 import javax.swing.*;
 import java.awt.*;
 
 public class MainPanel extends JPanel {
 
-    public MainPanel(MainFrameInterface frame) {
+    public MainPanel() {
```
Comment (Christina0313) [inline src/main/java/gui/MainPanel.java:9]: MainPanel takes MainFrameInterface as parameter. Please double check the main

File: src/main/java/entity/UserList.java (hunk 1 above)
Comment (Jennifer-xxx) [reply src/main/java/entity/UserList.java]: Deleted.

File: src/main/java/entity/UserListIterator.java (hunk 2 above)
Comment (Jennifer-xxx) [reply src/main/java/entity/UserListIterator.java]: Deleted.

File: src/main/java/entity/UserPreference.java (hunk 3 above)
Comment (Jennifer-xxx) [reply src/main/java/entity/UserPreference.java]: Deleted.

File: src/main/java/gui/MainPanel.java (hunk 6 above)
Comment (Jennifer-xxx) [reply src/main/java/gui/MainPanel.java]: Changed back.

**Review by imamberr — APPROVED** [review]
Chat feature is highly related with my branch. Well structure and nice UI make this feature really clear. Everything looks good and ready to merge.

**Review by yumozi — APPROVED** [review]
The code looks good and seems ready for merge.

**Review by alyssamyli — APPROVED** [review]
Alyssa reviewed and approved your pull request. Modification on UIs and connection to database is clean and efficient. Nice work

**Pull request merged**